lib/ts/recipe/passwordless/index.ts

```diff
@@ -0,0 +1,156 @@
+/* Copyright (c) 2022, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { getNormalisedUserContext } from "../../utils";
+import { RecipeFunctionOptions } from "../recipeModule/types";
+import Recipe from "./recipe";
+import {
+    InputType,
+    PasswordlessFlowType,
+    PasswordlessUser,
+    PostAPIHookContext,
+    PreAPIHookContext,
+    RecipeInterface,
+} from "./types";
+
+export default class RecipeWrapper {
+    static init(config?: InputType) {
+        return Recipe.init(config);
+    }
+
+    static createCode(
+        input:
+            | { email: string; userContext?: any; options?: RecipeFunctionOptions }
+            | { phoneNumber: string; userContext?: any; options?: RecipeFunctionOptions }
+    ): Promise<{
+        status: "OK";
+        deviceId: string;
+        preAuthSessionId: string;
+        flowType: PasswordlessFlowType;
+        fetchResponse: Response;
+    }> {
+        return Recipe.getInstanceOrThrow().recipeImplementation.createCode({
+            ...input,
+            userContext: getNormalisedUserContext(input.userContext),
+        });
+    }
+
+    static resendCode(input: {
+        deviceId: string;
+        preAuthSessionId: string;
+        userContext?: any;
+        options?: RecipeFunctionOptions;
+    }): Promise<{
+        status: "OK" | "RESTART_FLOW_ERROR";
+        fetchResponse: Response;
+    }> {
+        return Recipe.getInstanceOrThrow().recipeImplementation.resendCode({
+            ...input,
+            userContext: getNormalisedUserContext(input.userContext),
+        });
+    }
+
+    static consumeCode(
+        input:
+            | {
+                  userInputCode: string;
+                  deviceId: string;
+                  preAuthSessionId: string;
+                  userContext?: any;
+                  options?: RecipeFunctionOptions;
+              }
+            | {
+                  preAuthSessionId: string;
+                  linkCode: string;
```

Comment:
This can be fetched the link directly i guess?

Reply:
Same as above